lib/network-layer/src/Cardano/Wallet/Network/Streaming.hs

```diff
@@ -0,0 +1,165 @@
+{-# LANGUAGE DeriveFunctor #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Cardano.Wallet.Network.Streaming
+    ( withStreamingFromBlockChain
+    , TxWithContext
+    , OrRollback (..)
+    , FromBlockChain
+    , eraBlockS
+    , eraTxS
```

Comment:
This streaming interface is a little rough, but we can polish it later.

I don't think that the streaming interface should be the only interface to the chain synchronization — for example, the wallet uses `applyBlocks` directly, and the main advantage is that it's a pure function.

However, I see no problem with supporting `Cardano.Wallet.Network.Streaming` as an additional interface for people who like streaming — for some use cases, such as the present test, it's easy to use. 👍 Essentially, it provides value in the following ways: a) we avoid keeping track of a mutable state variable, and b) we can stop reading from the stream at any point in time. Both points are not provided by a callback mechanism.

(There are many streaming libraries, but I like `streaming`.)